File: ARRAY/Sorting_Implementation/Quick_sort.py

def quicksort(arr,low,high):
    if low < high:
        pivot_index = partition(arr,low,high)
        quicksort(arr,low,pivot_index-1)
        quicksort(arr,pivot_index+1,high)
        
        
def partition(arr, low , high):
    pivot = arr[low]
    i = low
    for j in range(low + 1 , high + 1):
        if arr[j] < pivot:
            i += 1
            arr[i], arr[j] = arr[j] , arr[i] # swap the array element with each other
    arr[low] , arr[i] = arr[i] , arr[low]
    
    return i



numbers = [7, 2, 9, 4, 1]
quicksort(numbers, 0, len(numbers) - 1)
print(numbers)

File: ARRAY/Sorting_Implementation/Quick_sort.js
function quickSort(arr, low, high) {
    if (low < high) {
        let pivot_index = partition(arr, low, high)
        quickSort(arr, low, pivot_index - 1)
        quickSort(arr, pivot_index + 1, high)
    }
}

function partition(arr, low, high) {
    let pivot = arr[low]
    let i = low
    for (let j = low + 1; j < high + 1; j++) {
        if (arr[j] < pivot) {
            i = i + 1
            let temp = arr[i]
            arr[i] = arr[j]
            arr[j] = temp
        }
    }
    let temp = arr[i]
    arr[i] = arr[low]
    arr[low] = temp
    return i
}

let numbers = [7, 2, 9, 4, 1]
quickSort(numbers, 0, 4)
console.log(numbers)   //   Quick sort does not return a array it just change the array 